### PR TITLE
Try running dev ubuntu image validation on GKE

### PR DESCRIPTION
**Pull request by yguo0905**
/assign @krzyzacy 
/cc @dashpole

### CODE REVIEW

**Comment by yguo0905** [comment]
/hold
until related GKE change gets submitted.

File: jobs/config.json [hunk 1]
```diff
@@ -3862,6 +3862,31 @@
...
+      "--cluster=test-118970819f",
+      "--env=CLOUDSDK_CORE_PRINT_UNHANDLED_TRACEBACKS=1",
+      "--extract=ci/latest",
+      "--gcp-node-image=CUSTOM",
+      "--gcp-project=ubuntu-os-gke-cloud-dev-tests",
+      "--gcp-zone=us-central1-f",
+      "--ginkgo-parallel=30",
+      "--image-family=ubuntu-gke-1604-lts-1",
+      "--image-project=ubuntu-os-gke-cloud-devel",
+      "--provider=gce",
```
Comment (dashpole) [inline jobs/config.json]: i think this needs to be gke

Comment (BenTheElder) [reply jobs/config.json]: this job is intentionally generated with GCE config, the job below is GKE. not sure if that makes sense given the context though..

Comment (yguo0905) [reply jobs/config.json]: Sorry, I should have clarified this. This is intentional. I'd like to create two jobs (gce and gke) and make sure the same `gcp-node-image` works on both platforms.

**Comment by dashpole** [comment]
looks like we need to whitelist the prow project for custom images

**Comment by dashpole** [comment]
and the GCE cluster creation doesnt like image type CUSTOM:
`Cannot operate on cluster using node os distro: CUSTOM`

**Comment by dashpole** [comment]
I added the visibility label to the prow project

**Comment by yguo0905** [comment]
The GKE one is failing because `--image-family` and `--image-project` didn't get passed through to gcloud.

Speaking of visibility, we need to whitelist all GKE projects in the pool.

**Comment by yguo0905** [comment]
The flags get consumed by the scenario script: https://github.com/kubernetes/test-infra/blob/master/scenarios/kubernetes_e2e.py#L590

I will move them into kubetest.

**Pull request merged**